shoes-core/lib/shoes/color/hex_converter.rb

```diff
@@ -26,8 +26,7 @@ def pad_if_necessary(hex)
       # Returns a 3- or 6-char hex string for valid input, or nil
       # for invalid input.
       def validate(hex)
-        match = /^#?(([0-9a-f]{3}){1,2})$/i.match(hex)
-        match && match[1]
+        /^#?(([0-9a-f]{3}){1,2})$/i.match(hex)&.[](1)
```

Comment:
This is the only line of this PR I don't like  - explicitly calling `[]` seems weird and perhaps unclear that it's equivalent

Alternatively, we could disable this cop on this line

Reply:
Agreed, that makes an already dense snippet even more obscure.

What's the cop say if we try to use a post-fix if instead?

```
       match = /^#?(([0-9a-f]{3}){1,2})$/i.match(hex)
       match[1] if match
```

Reply:
Could also move the regex into a constant imo :)

however yes, `&.[](1)` doesn't seem nice